mitosheet/mitosheet/utils.py

```diff
@@ -82,7 +82,61 @@ def is_default_df_names(df_names: List[str]) -> bool:
     """
     return len(df_names) > 0 and df_names == [f'df{i + 1}' for i in range(len(df_names))]
 
+
+def get_conditonal_formatting_result(
+        state: StateType,
+        sheet_index: int,
+        df: pd.DataFrame,
+        conditional_formatting_rules: List[Dict[str, Any]],
+        max_rows: Optional[int]=MAX_ROWS,
+    ) -> ConditionalFormattingResult: 
+
+    df = df.head(max_rows)
+
+    invalid_conditional_formats: ConditionalFormattingInvalidResults = dict()
+    formatted_result: ConditionalFormattingCellResults = dict()
+
+    for conditional_format in conditional_formatting_rules:
+        format_uuid = conditional_format["format_uuid"]
+        column_ids = conditional_format["columnIDs"]
+
+        for column_id in column_ids:
+            # Skip columns this is not applied to
+            if column_id not in column_ids:
+                continue
+            
+            if column_id not in formatted_result:
+                formatted_result[column_id] = dict()
+
+            filters  = conditional_format["filters"]
+            backgroundColor = conditional_format.get("backgroundColor", None)
+            color = conditional_format.get("color", None)
+
+            column_header = state.column_ids.get_column_header_by_id(sheet_index, column_id)
+
+            # Use the get_applied_filter function from our filtering infrastructure
+            from mitosheet.step_performers.filter import get_full_applied_filter
+            try:
+                full_applied_filter, _ = get_full_applied_filter(df, column_header, 'And', filters)
+                applied_indexes = full_applied_filter[full_applied_filter].index.tolist()
```

Comment:
This might be more clear as `df[full_applied_filter`] since we're applying the boolean index array to the dataframe to get the true indexes, right?